packages/dynamo-store/source/src/DynamoStoreSource.mts

```diff
@@ -274,65 +269,35 @@ function inflate(event: ITimelineEvent<EventBody>): ITimelineEvent {
 }
 
 export class DynamoStoreSource {
-  limiter: LimitFunction
-
+  private inner: TailingFeedSource
   constructor(
     private readonly index: AppendsIndex.Reader,
     private epochs: AppendsEpoch.Reader.Service,
     private readonly options: Omit<CreateOptions, "context">,
   ) {
-    this.limiter = pLimit(options.maxConcurrentStreams)
-  }
-
-  async handleTranche(
-    client: DynamoStoreSourceClient,
-    partition: AppendsPartitionId,
-    signal: AbortSignal,
-  ) {
-    let pos = Checkpoint.ofPosition(
-      await this.options.checkpoints.load(
-        this.options.groupName,
-        AppendsPartitionId.toString(partition),
-      ),
-    )
-    if (pos === Checkpoint.initial && this.options.startFromTail) {
-      pos = await Impl.readTailPositionForTranche(this.index, this.epochs, partition)
-    }
-    while (!signal.aborted) {
-      for await (const batch of client.crawl(partition, pos)) {
-        const streams = new Map<StreamName, ITimelineEvent[]>()
-        for (const [stream, event] of batch.items) {
-          const events = streams.get(stream)
-          if (events) {
-            events.push(inflate(event))
-          } else {
-            streams.set(stream, [inflate(event)])
-          }
-        }
-
-        const promises = []
-        for (const [stream, events] of streams) {
-          promises.push(this.limiter(() => this.options.handler(stream, events)))
-        }
-
-        await Promise.all(promises)
-        if (batch.checkpoint !== pos) {
-          await this.options.checkpoints.commit(
-            this.options.groupName,
-            AppendsPartitionId.toString(partition),
-            batch.checkpoint,
-          )
-          pos = batch.checkpoint
-        }
-        if (batch.isTail) await sleep(this.options.tailSleepIntervalMs, signal).catch(() => {})
-      }
+    if (!this.options.categories && !this.options.categoryFilter) {
```

Comment:
in cosmos I recently added a stream level filter - if there is no other usage of the category name in the lower level, better to map categories/categoryFilter/steamFilter to a stream predicate lambda at this level (though default surface API of a list of categories as a prominent default continues to be the main line)